proto/lbm/collection/v1/event.proto

```diff
@@ -0,0 +1,263 @@
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // name of the contract.
+  string name = 2;
+  // metadata of the contract.
+  string meta = 3;
+  // uri for the contract image stored off chain.
+  string base_img_uri = 4;
+}
+
+// EventCreatedFTClass is emitted when a new fungible token class is created.
+//
+// Since: 0.46.0 (finschia)
+message EventCreatedFTClass {
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // class id associated with the token class.
+  string class_id = 2;
+  // name of the token class.
+  string name = 3;
+  // metadata of the token class.
+  string meta = 4;
+  // decimals of the token class.
+  int32 decimals = 5;
+  // mintable represents whether the token class is allowed to mint or burn its tokens.
+  bool mintable = 6;
+}
+
+// EventCreatedNFTClass is emitted when a new non-fungible token class is created.
+//
+// Since: 0.46.0 (finschia)
+message EventCreatedNFTClass {
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // class id associated with the token class.
+  string class_id = 2;
+  // name of the token class.
+  string name = 3;
+  // metadata of the token class.
+  string meta = 4;
+}
+
+// EventGrant is emitted when a granter grants its permission to a grantee.
+//
+// Info: `granter` would be empty if the permission is granted by an issuance.
+//
+// Since: 0.46.0 (finschia)
+message EventGrant {
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // address of the granter which grants the permission.
+  string granter = 2;
+  // address of the grantee.
+  string grantee = 3;
+  // permission on the contract.
+  string permission = 4;
+}
+
+// EventAbandon is emitted when a grantee abandons its permission.
+//
+// Since: 0.46.0 (finschia)
+message EventAbandon {
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // address of the grantee which abandons its grant.
+  string grantee = 2;
+  // permission on the contract.
+  string permission = 3;
+}
+
+// EventMintedFT is emitted when fungible tokens are minted.
+//
+// Since: 0.46.0 (finschia)
+message EventMintedFT {
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // address which triggered the mint.
+  string operator = 2;
+  // recipient of the tokens.
+  string to = 3;
+  // amount of tokens minted.
+  repeated Coin amount = 4 [(gogoproto.nullable) = false];
+}
+
+// EventMintedNFT is emitted when non-fungible tokens are minted.
+//
+// Since: 0.46.0 (finschia)
+message EventMintedNFT {
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // address which triggered the mint.
+  string operator = 2;
+  // recipient of the tokens.
+  string to = 3;
+  // tokens minted.
+  repeated NFT tokens = 4 [(gogoproto.nullable) = false];
+}
+
+// EventBurned is emitted when tokens are burnt.
+//
+// Since: 0.46.0 (finschia)
+message EventBurned {
```

Comment:
`EventBurned` seems token's event. Isn't `EventFTBurned` and `EventNFTBurned` needed?

Reply:
No, like `EventSent`, you don't need to separate the both cases.

Reply:
I got it.
Another question) This does not have `token_id`. How do I know which FT or NFT is burned?

Reply:
> In case of Mint, There are separated `EventMintedFT` and `EventMintedNFT` but case of burned, only `EventBurned` exists. I feel it is unconsistent.

Sorry, I cannot merge `EventMintedFT` and `EventMintedNFT`, because we must minimize the changes in proto due to the schedule.



> Another question) This does not have `token_id`. How do I know which FT or NFT is burned?

`amount` field has the information in it.